packages/gradle/src/plugin/dependencies.ts

```diff
@@ -2,94 +2,49 @@ import {
   CreateDependencies,
   CreateDependenciesContext,
   DependencyType,
-  FileMap,
-  RawProjectGraphDependency,
+  StaticDependency,
   validateDependency,
+  workspaceRoot,
 } from '@nx/devkit';
-import { basename, dirname } from 'node:path';
+import { relative } from 'node:path';
 
-import { getCurrentGradleReport } from '../utils/get-gradle-report';
-import { GRADLE_BUILD_FILES } from '../utils/split-config-files';
+import { getCurrentProjectGraphReport } from './utils/get-project-graph-from-gradle-plugin';
 
 export const createDependencies: CreateDependencies = async (
   _,
   context: CreateDependenciesContext
 ) => {
-  const gradleFiles: string[] = findGradleFiles(context.filesToProcess);
-  if (gradleFiles.length === 0) {
-    return [];
-  }
-
-  const gradleDependenciesStart = performance.mark('gradleDependencies:start');
-  const {
-    gradleFileToGradleProjectMap,
-    gradleProjectNameToProjectRootMap,
-    buildFileToDepsMap,
-    gradleProjectToChildProjects,
-  } = getCurrentGradleReport();
-  const dependencies: Set<RawProjectGraphDependency> = new Set();
-
-  for (const gradleFile of gradleFiles) {
-    const gradleProject = gradleFileToGradleProjectMap.get(gradleFile);
-    const projectName = Object.values(context.projects).find(
-      (project) => project.root === dirname(gradleFile)
-    )?.name;
-    const dependedProjects: Set<string> = buildFileToDepsMap.get(gradleFile);
-
-    if (projectName && dependedProjects?.size) {
-      dependedProjects?.forEach((dependedProject) => {
-        const targetProjectRoot = gradleProjectNameToProjectRootMap.get(
-          dependedProject
-        ) as string;
-        const targetProjectName = Object.values(context.projects).find(
-          (project) => project.root === targetProjectRoot
-        )?.name;
-        if (targetProjectName) {
-          const dependency: RawProjectGraphDependency = {
-            source: projectName as string,
-            target: targetProjectName as string,
-            type: DependencyType.static,
-            sourceFile: gradleFile,
-          };
-          validateDependency(dependency, context);
-          dependencies.add(dependency);
-        }
-      });
-    }
-    gradleProjectToChildProjects.get(gradleProject)?.forEach((childProject) => {
-      if (childProject) {
-        const dependency: RawProjectGraphDependency = {
-          source: projectName as string,
-          target: childProject,
-          type: DependencyType.static,
-          sourceFile: gradleFile,
-        };
-        validateDependency(dependency, context);
-        dependencies.add(dependency);
+  const { dependencies: dependenciesFromReport } =
```

Comment:
here we just get the project graph report right away without calling `populateProjectGraph`.
I guess there are no cases where `createDependencies` will be called without `createNodes` being called before it?

Reply:
yes. plugin v1 has this behaviour, so createNodes will be called first and no need to call `populateProjectGraph`